src/update.cpp

```diff
@@ -279,6 +267,36 @@ void Update::run(int nsteps)
 
     // move particles
 
+    if (grid->uniform && enable_optmove)
```

Comment:
I think this should be set once before the run (e.g. in init), not every step, whcih is when the error or warning is issued.  As it is, this will be set even if surfs exist - won't tjhat be an error in comm.cpp ?

Reply:
@sjplimp the thought was fix adapt could change the grid from uniform to non-uniform, which could happen mid-run, so we need to check for that every timestep.

Reply:
I suppose we could just error out if fix adapt is active and a user enables optimized moves though.

Reply:
That's what I suggest.  I don't think we should mix opt move with grid adaptation.  If I understand it, opt move is for systems where particles uniformly fill a box.  No surfs, limited density variations.  If a user wants to create a coarse gird, then adapt it to a system with a lot of particles, they can do that in the first run, then enable opt move and do a second (equilibrium) run.